CHANGELOG.md

```diff
@@ -44,6 +44,10 @@ If possible, provide tooling that performs the changes, e.g. a shell-script.
 * Resolved an issue that prevented the FM-Index from being constructed correctly for inputs larger than 4 GiB
   ([\#2756](https://github.com/seqan/seqan3/pull/2756)).
 
+## API changes 
+
+The in 3.0.3 deprecated files (marked there with a `[deleted without replacement]`) have been removed.
```

Comment:
Maybe we could add a link to the `## API changes` section of 3.0.3?
I know we can do this but I don't remember how, I think we do doxygen style anchors?